data/txHandler.go

```diff
@@ -198,17 +204,36 @@ func (handler *TxHandler) asyncVerifySignature(arg interface{}) interface{} {
 }
 
 func (handler *TxHandler) processIncomingTxn(rawmsg network.IncomingMessage) network.OutgoingMessage {
-	var unverifiedTxn transactions.SignedTxn
-	err := protocol.Decode(rawmsg.Data, &unverifiedTxn)
-	if err != nil {
-		logging.Base().Warnf("Received a non-decodable txn: %v", err)
+	dec := protocol.NewDecoderBytes(rawmsg.Data)
+	ntx := 0
+	unverifiedTxGroup := make([]transactions.SignedTxn, 1)
+	for {
+		if len(unverifiedTxGroup) == ntx {
+			n := make([]transactions.SignedTxn, len(unverifiedTxGroup)*2)
```

Comment:
To be sure I'm not missing something subtle here: Instead of decoding and calling `append()`, you're growing the array yourself so that you can decode directly into `&unverifiedTxGroup[ntx]`, saving a memcpy, right? Or am I missing something?

Reply:
Yes, that's all that's going on here.

I believe the code was originally carefully optimized to the level where an extra copy of the `SignedTxn` mattered, so I tried to preserve that property.